airbyte-integrations/connectors/destination-iceberg/src/main/java/io/airbyte/integrations/destination/iceberg/config/format/FormatConfig.java

```diff
@@ -0,0 +1,59 @@
+package io.airbyte.integrations.destination.iceberg.config.format;
+
+import static io.airbyte.integrations.destination.iceberg.IcebergConstants.AUTO_COMPACT_CONFIG_KEY;
+import static io.airbyte.integrations.destination.iceberg.IcebergConstants.COMPACT_TARGET_FILE_SIZE_IN_MB_CONFIG_KEY;
+import static io.airbyte.integrations.destination.iceberg.IcebergConstants.FLUSH_BATCH_SIZE_CONFIG_KEY;
+import static io.airbyte.integrations.destination.iceberg.IcebergConstants.FORMAT_TYPE_CONFIG_KEY;
+import static io.airbyte.integrations.destination.iceberg.config.catalog.IcebergCatalogConfigFactory.getProperty;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.Data;
+
+/**
+ * @author Leibniz on 2022/10/31.
+ */
+@Data
+public class FormatConfig {
+
+    public static final int DEFAULT_FLUSH_BATCH_SIZE = 10000;
```

Comment:
Buffering records based on number is a bit risky since 10_000 records depending on size can quickly exhaust your memory. It is better if actual storage size is taken into consideration as well.